### PR TITLE
update docs to demonstrate best practices in regards to named templates

### PR DESCRIPTION
**Pull request by mueedc-zz**
This PR modified named template documentation:

update helper examples to demonstrate namespace best practices

highlight that templates names are global at the top of the documentation

address #3487
(opened a new PR in reference to #3510, I was not sure if I rebase/squashed correctly and did not want to mess with anything)

### CODE REVIEW

**Comment by mueedc** [comment]
@jascott1 @unguiculus I'd love any resources you could recommend on how to properly git rebase/squash commits

**Review by jascott1 — APPROVED** [review]

**Comment by jascott1** [comment]
@mueedc Rebase is worth understanding and the git-scm book covers this and more here [here](https://git-scm.com/book/id/v2/Git-Tools-Rewriting-History). If you want to learn git from the ground up I have found this to be helpful https://learngitbranching.js.org/

**Comment by jascott1** [comment]
BTW Since this is a large commit we will need another core to agree and merge.

cc @bacongobbler @SlickNik

**Comment by mueedc** [comment]
I have a base understanding of git but a refresher is never a bad idea. Thanks so much!

**Comment by bacongobbler** [comment]
looks like the indentation for the templates were accidentally changed from 2 spaces to 1 space during the rebase. Is there any way you could revert that back? :)

Also cc'ing @unguiculus in case he wants to give it a once-over one more time to make sure all his comments were addressed. :)

**Comment by jascott1** [comment]
Thanks @bacongobbler!

**Comment by mueedc** [comment]
I fixed the spacing issue, and I believe I rebase/squashed the commits down to 1. Hope I did that correctly

File: docs/chart_template_guide/named_templates.md [hunk 1]
```diff
@@ -48,36 +52,36 @@ apiVersion: v1
...
 metadata:
   name: {{ .Release.Name }}-configmap
-  {{- template "my_labels" }}
+  {{- template "mychart.labels" }}
 data:
-  myvalue: "Hello World"
-  {{- range $key, $val := .Values.favorite }}
-  {{ $key }}: {{ $val | quote }}
-  {{- end }}
+ myvalue: "Hello World"
```
Comment (jascott1) [inline docs/chart_template_guide/named_templates.md]: Needs another space as does following 3 lines.

File: docs/chart_template_guide/named_templates.md [hunk 2]
```diff
@@ -48,36 +52,36 @@ apiVersion: v1
...
 ```yaml
 # Source: mychart/templates/configmap.yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: running-panda-configmap
-  labels:
-    generator: helm
-    date: 2016-11-02
+ name: running-panda-configmap
```
Comment (jascott1) [inline docs/chart_template_guide/named_templates.md]: Fix spacing for this section too.

File: docs/chart_template_guide/named_templates.md [hunk 3]
```diff
@@ -93,25 +97,23 @@ apiVersion: v1
...
 metadata:
   name: {{ .Release.Name }}-configmap
-  {{- template "my_labels" }}
+  {{- template "mychart.labels" }}
 data:
-  myvalue: "Hello World"
-  {{- range $key, $val := .Values.favorite }}
-  {{ $key }}: {{ $val | quote }}
-  {{- end }}
+ myvalue: "Hello World"
```
Comment (jascott1) [inline docs/chart_template_guide/named_templates.md]: Fix spacing for this section.

File: docs/chart_template_guide/named_templates.md [hunk 4]
```diff
@@ -127,18 +129,18 @@ If we render this, the result will not be what we expect:
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: moldy-jaguar-configmap
-  labels:
-    generator: helm
-    date: 2016-11-02
-    chart:
-    version:
+ name: moldy-jaguar-configmap
```
Comment (jascott1) [inline docs/chart_template_guide/named_templates.md]: Fix spacing.

File: docs/chart_template_guide/named_templates.md [hunk 5]
```diff
@@ -160,12 +162,12 @@ Now when we execute this template with `helm install --dry-run --debug ./mychart
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: plinking-anaco-configmap
-  labels:
-    generator: helm
-    date: 2016-11-02
-    chart: mychart
-    version: 0.1.0
+ name: plinking-anaco-configmap
```
Comment (jascott1) [inline docs/chart_template_guide/named_templates.md]: Fix spacing.

File: docs/chart_template_guide/named_templates.md [hunk 6]
```diff
@@ -206,15 +207,15 @@ The output will not be what we expect:
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: measly-whippet-configmap
-  labels:
-    app_name: mychart
+ name: measly-whippet-configmap
```
Comment (jascott1) [inline docs/chart_template_guide/named_templates.md]: Fix spacing.

File: docs/chart_template_guide/named_templates.md [hunk 7]
```diff
@@ -246,16 +247,16 @@ Now the produced YAML is correctly indented for each section:
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: edgy-mole-configmap
-  labels:
-    app_name: mychart
-    app_version: "0.1.0+1478129987"
+ name: edgy-mole-configmap
```
Comment (jascott1) [inline docs/chart_template_guide/named_templates.md]: Fix spacing.

File: docs/chart_template_guide/named_templates.md [hunk 8]
```diff
@@ -228,15 +229,15 @@ Here's the example above, corrected to use `indent` to indent the `mychart_app`
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: {{ .Release.Name }}-configmap
-  labels:
-{{ include "mychart_app" . | indent 4 }}
+ name: {{ .Release.Name }}-configmap
```
Comment (jascott1) [inline docs/chart_template_guide/named_templates.md]: Fix spacing.

**Review by jascott1 — CHANGES_REQUESTED** [review]
Please fix the indentations. All YAML should use 2 spaces of indentation for each level of nested keys.

**Comment by jascott1** [comment]
@mueedc You are looking for `git rebase -i master` on your branch. Then you change all your commits but the first from `pick` to `squash` and save (deleting the extra commit messages on the subsequent screens). Finally you push to your repo with a `--force`. Hope that helps.

**Comment by mueedc** [comment]
@jascott1 thanks for that. I think I got the hang of rebasing now and I believe everything should be good for the merge. @unguiculus this should address your comments as well.

**Review by jascott1 — APPROVED** [review]

**Comment by jascott1** [comment]
@unguiculus I think its good 2 go but will wait for your review. Thanks!

**Comment by mueedc** [comment]
Thanks for all the pointers along the way @jascott1. This has been an immensely helpful learning experience!

**Comment by jascott1** [comment]
@mueedc Glad I could help. Thanks for the contribution!

**Pull request merged**